Rlabkey/R/labkey.pipeline.R

```diff
@@ -0,0 +1,135 @@
+    if (missing(baseUrl) || is.null(baseUrl)) stop (paste("A value must be specified for baseUrl."))
+    if (missing(folderPath)) stop (paste("A value must be specified for folderPath."))
+
+    ## normalize the folder path
+    folderPath <- encodeFolderPath(folderPath)
+
+    url <- paste(baseUrl, "pipeline", folderPath, "getPipelineContainer.api", sep="")
+    response <- labkey.get(url)
+
+    return (fromJSON(response, simplifyVector=FALSE, simplifyDataFrame=FALSE))
+}
+
+labkey.pipeline.getProtocols <- function(baseUrl=NULL, folderPath, taskId, path, includeWorkbooks = FALSE)
+{
+    baseUrl=labkey.getBaseUrl(baseUrl)
+
+    ## check required parameters
+    if (missing(baseUrl) || is.null(baseUrl)) stop (paste("A value must be specified for baseUrl."))
+    if (missing(folderPath)) stop (paste("A value must be specified for folderPath."))
+    if (missing(taskId) || is.null(taskId)) stop (paste("A value must be specified for taskId."))
+    if (missing(path) || is.null(path)) stop (paste("A value must be specified for path."))
+
+    ## normalize the folder path
+    folderPath <- encodeFolderPath(folderPath)
+
+    params <- list(taskId = taskId, path = path, includeWorkbooks = includeWorkbooks)
+
+    url <- paste(baseUrl, "pipeline-analysis", folderPath, "getSavedProtocols.api", sep="")
+    response <- labkey.post(url, toJSON(params, auto_unbox=TRUE))
+
+    return (fromJSON(response, simplifyVector=FALSE, simplifyDataFrame=FALSE))
+}
+
+labkey.pipeline.getFileStatus <- function(baseUrl=NULL, folderPath, taskId, protocolName, path, files)
+{
+    baseUrl=labkey.getBaseUrl(baseUrl)
+
+    ## check required parameters
+    if (missing(baseUrl) || is.null(baseUrl)) stop (paste("A value must be specified for baseUrl."))
+    if (missing(folderPath)) stop (paste("A value must be specified for folderPath."))
+    if (missing(taskId) || is.null(taskId)) stop (paste("A value must be specified for taskId."))
+    if (missing(protocolName) || is.null(protocolName)) stop (paste("A value must be specified for protocolName."))
+    if (missing(path) || is.null(path)) stop (paste("A value must be specified for path."))
+    if (missing(files) || is.null(files)) stop (paste("A value must be specified for files."))
+
+    ## check parameter types
+    if (!is.list(files)) stop (paste("The files parameter must be a list of strings."))
+
+    ## normalize the folder path
+    folderPath <- encodeFolderPath(folderPath)
+
+    params <- list(taskId = taskId, protocolName = protocolName, path = path, file = files)
+
+    url <- paste(baseUrl, "pipeline-analysis", folderPath, "getFileStatus.api", sep="")
+    response <- labkey.post(url, toJSON(params, auto_unbox=TRUE))
+
+    return (fromJSON(response, simplifyVector=FALSE, simplifyDataFrame=FALSE))
+}
+
+labkey.pipeline.startAnalysis <- function(baseUrl=NULL, folderPath, taskId, protocolName, path, files,
+    fileIds = list(), pipelineDescription = NULL, protocolDescription = NULL,
```

Comment:
by providing a default value for fileIds, does this mean that it becomes optional and the check on line 97 would never be hit?

Reply:
Yes, the default value provided does make it optional, however the check at line 97 is for if the user does decide to pass in this parameter and they pass in a non-list type variable.